proto/ibc/core/channel/v1/event.proto

```diff
@@ -0,0 +1,210 @@
+syntax = "proto3";
+package ibc.core.channel.v1;
+
+option go_package = "github.com/cosmos/cosmos-sdk/x/ibc/core/04-channel/types";
+
+import "ibc/core/channel/v1/channel.proto";
+import "gogoproto/gogo.proto";
+import "ibc/core/client/v1/client.proto";
+
+// EventChannelOpenInit is a typed event emitted on channel open init
+message EventChannelOpenInit {
+    string port_id = 1 [(gogoproto.moretags) = "yaml:\"port_id\""];
+
+    string channel_id = 2 [(gogoproto.moretags) = "yaml:\"channel_id\""];
+
+    string counterparty_port_id = 3 [(gogoproto.moretags) = "yaml:\"counterparty_port_id\""];
+
+    string counterparty_channel_id = 4 [(gogoproto.moretags) = "yaml:\"counterparty_channel_id\""];
+
+    string connection_id = 5 [(gogoproto.moretags) = "yaml:\"connection_id\""];
+}
+
+// EventChannelOpenTry is a typed event emitted on channel open try
+message EventChannelOpenTry {
+    string port_id = 1 [(gogoproto.moretags) = "yaml:\"port_id\""];
+
+    string channel_id = 2 [(gogoproto.moretags) = "yaml:\"channel_id\""];
+
+    string counterparty_port_id = 3 [(gogoproto.moretags) = "yaml:\"counterparty_port_id\""];
+
+    string counterparty_channel_id = 4 [(gogoproto.moretags) = "yaml:\"counterparty_channel_id\""];
+
+    string connection_id = 5 [(gogoproto.moretags) = "yaml:\"connection_id\""];
+}
+
+// EventChannelOpenAck is a typed event emitted on channel open acknowledgement
+message EventChannelOpenAck {
+    string port_id = 1 [(gogoproto.moretags) = "yaml:\"port_id\""];
+
+    string channel_id = 2 [(gogoproto.moretags) = "yaml:\"channel_id\""];
+
+    string counterparty_port_id = 3 [(gogoproto.moretags) = "yaml:\"counterparty_port_id\""];
+
+    string counterparty_channel_id = 4 [(gogoproto.moretags) = "yaml:\"counterparty_channel_id\""];
+
+    string connection_id = 5 [(gogoproto.moretags) = "yaml:\"connection_id\""];
+}
+
+// EventChannelCloseInit is a typed event emitted on channel close init
+message EventChannelCloseInit {
```

Comment:
nit: reorder so `OpenConfirm` comes before this. Open handshake steps should be grouped together